From: Julian Wiedmann <jwi@linux.vnet.ibm.com>
Subject: s390/qeth: add missing hash table initializations
Patch-mainline: v4.12-rc2
Git-commit: ebccc7397e4a49ff64c8f44a54895de9d32fe742
References: bnc#1072216, LTC#162173

Description:  qeth: fixes for L3 set_rx_mode improvements
Symptom:      The L3 set_rx_mode improvements introduced some
              problems that can result in access of uninitialized
              memory, a blocked refcount and intermittent loss
              of inbound IPv4 multicast traffic.
Problem:      see individual patch descriptions
Solution:     Apply all relevant fixes.
Reproduction: -

Upstream-Description:

              s390/qeth: add missing hash table initializations

              commit 5f78e29ceebf ("qeth: optimize IP handling in rx_mode callback")
              added new hash tables, but missed to initialize them.

              Fixes: 5f78e29ceebf ("qeth: optimize IP handling in rx_mode callback")
              Signed-off-by: Ursula Braun <ubraun@linux.vnet.ibm.com>
              Reviewed-by: Julian Wiedmann <jwi@linux.vnet.ibm.com>
              Signed-off-by: Julian Wiedmann <jwi@linux.vnet.ibm.com>
              Signed-off-by: David S. Miller <davem@davemloft.net>


Signed-off-by: Julian Wiedmann <jwi@linux.vnet.ibm.com>
Acked-by: Johannes Thumshirn <jthumshirn@suse.de>
---
 drivers/s390/net/qeth_l3_main.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/s390/net/qeth_l3_main.c
+++ b/drivers/s390/net/qeth_l3_main.c
@@ -3254,6 +3254,8 @@ static int qeth_l3_probe_device(struct c
 	rc = qeth_l3_create_device_attributes(&gdev->dev);
 	if (rc)
 		return rc;
+	hash_init(card->ip_htable);
+	hash_init(card->ip_mc_htable);
 	card->options.layer2 = 0;
 	card->info.hwtrap = 0;
 	return 0;
